.github/workflows/ci.yml

```diff
@@ -312,7 +312,7 @@ jobs:
       fail-fast: false
       matrix:
         # test the oldest supported version and main
-        typing-extensions-version: ['4.6.1', 'main']
+        typing-extensions-version: ['4.9.0', 'main']
```

Comment:
I would suggest undoing this change for reasons noted in https://github.com/pydantic/pydantic/pull/8237/files#r1424563069